arch/arm/boot/dts/broadcom/bcm2711-rpi-ds.dtsi

```diff
@@ -147,18 +147,11 @@
 		interrupts = <GIC_SPI 176 IRQ_TYPE_LEVEL_HIGH>;
 		power-domains = <&power RPI_POWER_DOMAIN_USB>;
 	};
+};
 
-	codec@7eb10000 {
-		compatible = "raspberrypi,rpivid-vid-decoder";
-		reg = <0x0 0x7eb10000  0x0 0x1000>,  /* INTC */
-		      <0x0 0x7eb00000  0x0 0x10000>; /* HEVC */
-		reg-names = "intc",
-			    "hevc";
-		interrupts = <GIC_SPI 98 IRQ_TYPE_LEVEL_HIGH>;
-
-		clocks = <&firmware_clocks 11>;
-		clock-names = "hevc";
-	};
+&hevc_dec {
+	reg = <0x0 0x7eb10000  0x0 0x1000>,  /* INTC */
+	      <0x0 0x7eb00000  0x0 0x10000>; /* HEVC */
```

Comment:
INTC and HEVC are swapped compared to bcm2711.dtsi - this breaks HEVC on RPi4.

likely the hevc_dec override should just be dropped as reg is set in bcm2711.dtsi anyways

Reply:
Thanks.
Yes it is needed as `size-cells` differs between upstream and downstream, but the swapping is wrong.